test/run_test.py

```diff
@@ -25,6 +26,125 @@
+    os_env["PYTORCH_TEST_SKIP_CUDA_MEM_LEAK_CHECK"] = "1"
+else:
+    os_env["PYTORCH_TEST_SKIP_CUDA_MEM_LEAK_CHECK"] = "0"
+
+if "rocm" in BUILD_ENVIRONMENT:
+    # Print GPU info
+    shell(r"rocminfo | grep -E 'Name:.*\sgfx|Marketing'".split())
+
+if "ppc64le" not in BUILD_ENVIRONMENT and "-bazel-" not in BUILD_ENVIRONMENT:
+    # JIT C++ extensions require ninja.
+    shell("pip install --user ninja".split())
+    # ninja is installed in $HOME/.local/bin, e.g., /var/lib/jenkins/.local/bin for CI user jenkins
+    # but this script should be runnable by any user, including root
+    os_env["PATH"] = os_env.get("HOME", "") + "/.local/bin:" + os_env.get("PATH", "")
+
+if "asan" in BUILD_ENVIRONMENT:
+    # Suppress vptr violations arising from multiple copies of pybind11
+    os_env["ASAN_OPTIONS"] = "detect_leaks=0:symbolize=1:strict_init_order=true:detect_odr_violation=0"
+    os_env["UBSAN_OPTIONS"] = "print_stacktrace=1:suppressions=" + os_env.get("PWD", "") + "/ubsan.supp"
+    os_env["PYTORCH_TEST_WITH_ASAN"] = "1"
+    os_env["PYTORCH_TEST_WITH_UBSAN"] = "1"
+    # TODO: Figure out how to avoid hard-coding these paths
+    os_env["ASAN_SYMBOLIZER_PATH"] = "/usr/lib/llvm-7/bin/llvm-symbolizer"
+    os_env["TORCH_USE_RTLD_GLOBAL"] = "1"
+    # NB: We load libtorch.so with RTLD_GLOBAL for UBSAN, unlike our
+    # default behavior.
+    os_env["LD_PRELOAD"] = "/usr/lib/llvm-7/lib/clang/7.1.0/lib/linux/libclang_rt.asan-x86_64.so"
+
+if "-NO_AVX-" in BUILD_ENVIRONMENT or TEST_CONFIG == "nogpu_NO_AVX":
+    os_env["ATEN_CPU_CAPABILITY"] = "default"
+elif "-NO_AVX2-" in BUILD_ENVIRONMENT or TEST_CONFIG == "nogpu_NO_AVX2":
+    os_env["ATEN_CPU_CAPABILITY"] = "default"
+elif "-NO_AVX512-" in BUILD_ENVIRONMENT or TEST_CONFIG == "nogpu_NO_AVX512":
+    os_env["ATEN_CPU_CAPABILITY"] = "avx2"
+
+def print_to_stderr(message):
+    print(message, file=sys.stderr)
+
+def setup_test_aten():
+    if "asan" not in BUILD_ENVIRONMENT and "rocm" not in BUILD_ENVIRONMENT:
+        print_to_stderr("Setting up env for ATen tests")
+
+        if IN_WHEEL_TEST:
+            print_to_stderr("Using install folder")
+            # Rename the build folder when running test to ensure it
+            # is not depended on the folder
+            shell(("mv " + BUILD_DIR + " " + BUILD_RENAMED_DIR).split())
+            TEST_BASE_DIR = TORCH_TEST_DIR
+        else:
+            print_to_stderr("Using build folder")
+            TEST_BASE_DIR = BUILD_BIN_DIR
+
+        SUDO = "sudo" if "ppc64le" in BUILD_ENVIRONMENT else ""
+        shell((SUDO + " ln -sf " + TORCH_LIB_DIR + "/libc10* " + TEST_BASE_DIR).split())
+        shell((SUDO + " ln -sf " + TORCH_LIB_DIR + "/libcaffe2* " + TEST_BASE_DIR).split())
+        shell((SUDO + " ln -sf " + TORCH_LIB_DIR + "/libmkldnn* " + TEST_BASE_DIR).split())
+        shell((SUDO + " ln -sf " + TORCH_LIB_DIR + "/libnccl* " + TEST_BASE_DIR).split())
+        shell((SUDO + " ln -sf " + TORCH_LIB_DIR + "/libtorch* " + TEST_BASE_DIR).split())
```

Comment:
Please don't do that, but instead just use `LD_LIBRARY_PATH`
```suggestion
     os.env["LD_LIBRARY_PATH"] = f"{TORCH_LIB_DIR};{os.env.get('LD_LIBRARY_PATH',''}"
```